src/js/views/metadataView/DatasetControlsView.js

```diff
@@ -0,0 +1,905 @@
+            this.removeButton(name);
+            // eslint-disable-next-line no-console
+            console.error(`Error rendering the ${name} button:`, error);
+          }
+        });
+        return this;
+      },
+
+      /**
+       * Clear the view content and remove subviews and event listeners.
+       * @param {boolean} [destroyCitationModal] If true, also destroy the
+       * citation modal subview. We avoid this unless the entire view is being
+       * destroyed because the modal is modified externally (see renderCite()).
+       */
+      reset(destroyCitationModal = false) {
+        // Remove all the subviews
+        this.subviews.forEach((subview) => {
+          if (typeof subview.onClose === "function") {
+            subview.onClose();
+          }
+          subview.remove();
+        });
+        this.subviews = [];
+
+        // Remove all the buttons
+        Object.entries(this.buttons).forEach(([name]) =>
+          this.removeButton(name),
+        );
+
+        // Remove and close all subviews
+        if (destroyCitationModal && this.citationModal) {
+          this.citationModal.onClose();
+          this.citationModal.remove();
+          this.citationModal = null;
+        }
+        this.stopListening();
+      },
+
+      /**
+       * Get the container element where a button should be inserted.
+       * @param {string} name - Logical button name.
+       * @returns {HTMLElement|null} The container element or null if not found.
+       */
+      getButtonContainer(name) {
+        const configedButton = this.buttons[name];
+        if (!configedButton) return null;
+        const containerClass = configedButton.container;
+        return this.el.querySelector(`.${containerClass}`);
+      },
+
+      /**
+       * Get a previously rendered button element by name.
+       * @param {string} name - Logical button name.
+       * @returns {HTMLElement|null} The button element if it exists.
+       */
+      getExistingButtonEl(name) {
+        return this.buttonEls?.[name] || null;
+      },
+
+      /**
+       * Remove a button and clear its container.
+       * @param {string} name - Logical button name to remove.
+       */
+      removeButton(name) {
+        const buttonEl = this.getExistingButtonEl(name);
+        const container = this.getButtonContainer(name);
+        if (buttonEl) {
+          $(buttonEl).popup("destroy");
+          buttonEl.remove();
+          delete this.buttonEls?.[name];
+        }
+        if (container) {
+          container.innerHTML = "";
+          container.style.display = "none";
+        }
+      },
+
+      /**
+       * Get or create the metrics model used to fetch dataset metrics.
+       * @returns {MetricsModel} The metrics model instance for this view.
+       */
+      getMetricsModel() {
+        const model = this.viewModel.get("metricsModel");
+
+        // Check if the model's been fetched yet
+        if (!model.get("fetched") && !model.get("fetching")) {
```

Comment:
We do not have a fetched flag in the metrics model
```suggestion
        if (!model.get("synced") && !model.get("fetching")) {
```

Reply:
Great catch!! Thank you!!